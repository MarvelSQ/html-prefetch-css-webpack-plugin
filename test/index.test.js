const webpack = require("webpack");
const config = require("./webpack.config");
const fs = require("fs");
const path = require("path");
const cp = require("child_process");

cp.spawnSync("npm", ["run", "dev"]);

test("index html", () => {
  const buildFile = fs.readFileSync(
    path.resolve(__dirname, "dist/index.html"),
    {
      encoding: "utf-8"
    }
  );
  expect(buildFile).toMatchSnapshot();
});

test('second html', () => {
  const buildFile = fs.readFileSync(
    path.resolve(__dirname, "dist/second.html"),
    {
      encoding: "utf-8"
    }
  );
  expect(buildFile).toMatchSnapshot();
})
