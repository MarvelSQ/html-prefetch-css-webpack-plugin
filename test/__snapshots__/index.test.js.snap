// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index html 1`] = `
"<!DOCTYPE html>
<html lang=\\"en\\">
<head>
  <meta charset=\\"UTF-8\\">
  <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\">
  <meta http-equiv=\\"X-UA-Compatible\\" content=\\"ie=edge\\">
  <title>Html prefetch css webpack plugin</title>
<link href=\\"moduleA.41e02ef82460f1bbbe03.css\\" rel=\\"prefetch\\"><link href=\\"moduleB.41e02ef82460f1bbbe03.css\\" rel=\\"prefetch\\"><link href=\\"index.41e02ef82460f1bbbe03.css?41e02ef82460f1bbbe03\\" rel=\\"stylesheet\\"></head>
<body>
  <div id=\\"root\\"></div>
<script type=\\"text/javascript\\" src=\\"index.bundle.js?41e02ef82460f1bbbe03\\"></script></body>
</html>"
`;

exports[`second html 1`] = `
"<!DOCTYPE html>
<html lang=\\"en\\">
<head>
  <meta charset=\\"UTF-8\\">
  <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\">
  <meta http-equiv=\\"X-UA-Compatible\\" content=\\"ie=edge\\">
  <title>Html prefetch css webpack plugin</title>
<link href=\\"moduleC.41e02ef82460f1bbbe03.css\\" rel=\\"prefetch\\"></head>
<body>
  <div id=\\"root\\"></div>
<script type=\\"text/javascript\\" src=\\"second.bundle.js?41e02ef82460f1bbbe03\\"></script></body>
</html>"
`;
